scripts/pylib/pytest-twister-harness/src/twister_harness/multi_device_fixtures.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# Copyright (c) 2025 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+import logging
+import time
+from pathlib import Path
+
+import pytest
+from twisterlib.hardwaremap import HardwareMap
+
+from twister_harness.device.device_adapter import DeviceAdapter
+from twister_harness.device.hardware_adapter import HardwareAdapter
+from twister_harness.fixtures import determine_scope
+from twister_harness.helpers.shell import Shell
+from twister_harness.helpers.utils import find_in_config
+from twister_harness.twister_harness_config import DeviceConfig, TwisterHarnessConfig
+
+logger = logging.getLogger(__name__)
+
+
+class TwisterOptionsWrapper:
+    """Wrapper class for twister options"""
+
+    device_flash_timeout: float = 60.0  # [s]
+    device_flash_with_test: bool = True
+    flash_before: bool = False
+
+
+class TwisterEnvWrapper:
+    """Wrapper class for twister environment options"""
+
+    def __init__(self, options: TwisterOptionsWrapper):
+        self.options = options
+
+
+class HarnessHardwareAdapter(HardwareAdapter):
+    """Adapter class for real device."""
+
+    def __init__(self, device_config: DeviceConfig) -> None:
+        super().__init__(device_config)
+
+        self.device_log_path: Path = device_config.build_dir / f"device_{device_config.id}.log"
+        self.handler_log_path: Path = device_config.build_dir / f"handler_{device_config.id}.log"
+
+        self._log_files: list[Path] = [self.handler_log_path, self.device_log_path]
+
+
+@pytest.fixture(scope=determine_scope)
+def harness_devices(
+    request: pytest.FixtureRequest,
+    dut: DeviceAdapter,
+    twister_harness_config: TwisterHarnessConfig,
+    harness_build_dirs: list[str],
```

Comment:
where is `harness_build_dirs`? why it is defined in tests and used in a common fixture?